db/migrate/20180130160640_add_unique_constraint_to_entries.rb

```diff
@@ -0,0 +1,5 @@
+class AddUniqueConstraintToEntries < ActiveRecord::Migration[5.1]
+  def change
+    add_index(:entries, %i[hash_value entry_type timestamp spina_register_id], unique: true, name: 'unique_entry_index')
```

Comment:
Technically you could have multiple entries created at the same time for the same register and which point at the same object. The index should instead be on `[hash_value entry_type entry_number spina_register_id]`, as `entry_number` is guaranteed to be unique for each `entry_type`.

Reply:
A bit late but, I'm missing `key` here. If `key` is part of the index, isn't unique enough to make `timestamp` relevant?